controllers/controllers.go

```diff
@@ -187,3 +184,15 @@ func isValidWindowsNode(o client.Object, byoh bool) bool {
 	}
 	return true
 }
+
+// isWindowsNodeWithVersionMismatch returns true if the node object is a valid Windows node and if there is a
+// version mismatch which will require the node to be updated.
+func isWindowsNodeWithVersionMismatch(o client.Object, byoh bool) bool {
+	if !isValidWindowsNode(o, byoh) {
+		return false
+	}
+	if o.GetAnnotations()[metadata.VersionAnnotation] != version.Get() {
+		return true
+	}
+	return false
```

Comment:
Can this function be reduced to `return isValidWindowsNode(o, byoh) && o.GetAnnotations()[metadata.VersionAnnotation] != version.Get()`? And if so, is there a need to keep this in a helper? I am fine either way, just floating the idea.

Reply:
I am fine removing the helper.

Reply:
+1 not sure we need another helper, rest LGTM.

Reply:
Addressed. PTAL.

Reply:
The PR description needs an update removing first bullet point.

Reply:
Addressed. PTAL.

Reply:
/lgtm

Reply:
LGTM